### PR TITLE
Improve examples.

### PR DESCRIPTION
**Pull request by jan-ivar**
***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/webtransport/pull/182.html" title="Last updated on Jan 23, 2021, 1:28 AM UTC (b76e31a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webtransport/182/b8d1a94...jan-ivar:b76e31a.html" title="Last updated on Jan 23, 2021, 1:28 AM UTC (b76e31a)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -945,43 +949,80 @@ Receiving datagrams can be achieved by calling
...
 <pre class="example" highlight="js">
-const transport = getTransport();
-const reader = transport.datagramReadable.getReader();
-while (true) {
-  const {value: datagram, done} = await reader.read();
-  if (done) {
-    break;
+async function(url) {
+  const transport = new WebTransport(url);
+  for await (datagram of transport.datagramReadable) {
```
Comment (ricea) [inline index.bs:954]: I suggest writing `const datagram` to avoid using a global variable. Also below.

File: index.bs [hunk 2]
```diff
@@ -909,13 +909,14 @@ Sending a buffer of datagrams can be achieved by using the
...
 
 <pre class="example" highlight="js">
-const transport = getTransport();
-const writer = transport.datagramWritable.getWriter();
-const datagrams = getDatagramsToSend();
-datagrams.forEach((datagram) => {
-  await writer.ready;
-  writer.write(datagram);
-});
+async function(url, datagrams) {
```
Comment (yutakahirano) [inline index.bs:912]: It's better to name this function, or run this function. Ditto below.

File: index.bs (hunk 1 above)
Comment (jan-ivar) [reply index.bs:954]: Good catch, I keep doing that.

**Comment by jan-ivar** [comment]
Hmm, I added some error propagation for the in parallel steps, but I'm not happy with it, since it defers errors until the end. Let me try again.

**Comment by jan-ivar** [comment]
Are errors likely to occur per stream, or are they all fatal to the transport?

**Comment by jan-ivar** [comment]
Ended up reporting per stream, since the iteration on `incomingUnidirectionalStreams` itself should fail if the connection fails.

**Comment by jan-ivar** [comment]
For those interested, I wrote a quick [polyfill](https://jsfiddle.net/jib1/52byh1kq/) to verify this syntax:
```js
for await (const stream of transport.incomingUnidirectionalStreams) {
  await stream.readable.pipeThrough(new TextDecoderStream("utf-8")).pipeTo(new WritableConsole());
}
```
I dunno if it's production ready or anything, but it appears to work right in Canary.

**Comment by jan-ivar** [comment]
> ```js
> for await (const stream of transport.incomingUnidirectionalStreams) {
>  await stream.readable.pipeThrough(new TextDecoderStream("utf-8")).pipeTo(new WritableConsole());
>}
> ```

For those curious, the reason for not reading in parallel here is to avoid interleaving the text output, since it's going to a combined log.

Now that I think about it, that might be good to show in the text decoder example.

File: index.bs [hunk 3]
```diff
@@ -998,167 +1039,123 @@ the server, and sending and receiving datagrams.
...
-  entry.className = 'log-' + severity;
+  entry.className = &#96;log-${severity}&#96;;
   log.appendChild(entry);
 
-  // If the most recent entry in the log was visible, scroll the log to the
-  // newly added element.
-  if (mostRecentEntry != null &&
-      mostRecentEntry.getBoundingClientRect().top <
-          log.getBoundingClientRect().bottom) {
+  // If the last entry in the log is visible, scroll the log to the new element.
```
Comment (wilaw) [inline index.bs:1160]: If the most recent entry in the log is not visible, ...

Comment (jan-ivar) [reply index.bs:1160]: I renamed it to `last` for brevity in line 1150 above, otherwise the logic is unchanged, except I see present tense snuck in there, which doesn't help. Let me find a better name...

File: index.bs [hunk 4]
```diff
@@ -998,167 +1045,121 @@ the server, and sending and receiving datagrams.
...
-            'Opened bidirectional stream #' + number +
-            ' with data: ' + rawData);
+        await writer.close();
+        addToEventLog(&#96;Sent bidirectional stream #${n} with data: ${rawData}&#96;);
         break;
       }
     }
   } catch (e) {
-    addToEventLog('Error while sending data: ' + e, 'error');
+    addToEventLog(&#96;Error while sending data: ${e}&#96;, 'error');
```
Comment (martinthomson) [inline index.bs:1107]: We can't use `<![CDATA[` blocks so that this quoting is not necessary?

**Review by yutakahirano — APPROVED** [review]
